cpp/cmake_modules/BuildUtils.cmake

```
@@ -97,7 +97,30 @@ function(arrow_create_merged_static_lib output_target)
   endforeach()
 
   if(APPLE)
-    set(BUNDLE_COMMAND "libtool" "-no_warning_for_no_symbols" "-static" "-o"
+    # The apple-distributed libtool is what we want for bundling, but there is
+    # a GNU libtool that has a namecollision (and happens to be bundled with R, too).
+    # We are not compatible with GNU libtool, so we need to avoid it.
+    # TODO: use a VALIDATOR when we require cmake >= 3.25
+
+    # check in the obvious places first to find Apple's libtool
+    find_program(LIBTOOL_MACOS libtool
+                 PATHS /usr/bin /Library/Developer/CommandLineTools/usr/bin
+                 NO_DEFAULT_PATH)
+
+    # if it's not found in obvious places, check on the standard path
+    if(LIBTOOL_MACOS-NOTFOUND)
+      set(LIBTOOL_MACOS "libtool")
+    endif()
```

How about just using default paths? `$PATH` is used after `PATHS`.

```suggestion
                 PATHS /usr/bin /Library/Developer/CommandLineTools/usr/bin)
```

https://cmake.org/cmake/help/latest/command/find_program.html

> 4. Search the paths specified by the HINTS option. These should be paths computed by system introspection, such as a hint provided by the location of another item already found. Hard-coded guesses should be specified with the PATHS option.
>
> 5. Search the standard system environment variables. This can be skipped if NO_SYSTEM_ENVIRONMENT_PATH is passed or by setting the [CMAKE_FIND_USE_SYSTEM_ENVIRONMENT_PATH](https://cmake.org/cmake/help/latest/variable/CMAKE_FIND_USE_SYSTEM_ENVIRONMENT_PATH.html#variable:CMAKE_FIND_USE_SYSTEM_ENVIRONMENT_PATH) to FALSE.
>
>    * The directories in PATH itself.

---

I can try this. I will admit I read that list in the find docs a few times and was not 100% certain it would follow the path I was hoping it would (use our provided paths first, then project paths, then system paths).

---

Ok, this works though I had to switch from `PATHS` to `HINTS` since `HINTS` is higher in the search. This is a bit funny given the docs:

> 4. Search the paths specified by the HINTS option. These should be paths computed by system introspection, such as a hint provided by the location of another item already found. Hard-coded guesses should be specified with the PATHS option.

But it does seem to work!